examples/js/lines/LineMaterial.js

```diff
@@ -205,19 +205,19 @@ THREE.ShaderLib[ 'line' ] = {
 
 			#ifdef USE_DASH
 
-				if ( vUv.y < 0.5 || vUv.y > 0.5 ) discard; // discard endcaps
+				if ( vUv.y < - 1.0 || vUv.y > 1.0 ) discard; // discard endcaps
```

Comment:
It has a lot of checks though:
```
2x ||
1x ?
4x >
2x if
```

By adding another dimension, you can do it with `bool isCap = vUv.z < 0.5;` and two `if` checks.